odoo/addons/base/models/ir_mail_server.py

```diff
@@ -143,6 +143,46 @@ def _check_smtp_ssl_files(self):
             elif mail_server.smtp_ssl_private_key and not mail_server.smtp_ssl_certificate:
                 raise UserError(_('SSL certificate is missing for %s.', mail_server.name))
 
+    def write(self, vals):
+        """Ensure we cannot archive a server in-use"""
+        has_archive_error = False
```

Comment:
Note that ``has_archive_error`` is not necessary, could directly use ``usages`` initialized to a void list by default :)

Reply:
Yes that's simpler, changed